src/main/java/software/amazon/encryption/s3/materials/KmsKeyring.java

```diff
@@ -102,6 +107,32 @@ public EncryptionMaterials modifyMaterials(EncryptionMaterials materials) {
                     .build();
         }
 
+        @Override
+        public EncryptionMaterials generateDataKey(EncryptionMaterials materials) {
+            GenerateDataKeyRequest request = GenerateDataKeyRequest.builder()
+                    .keyId(_wrappingKeyId)
+                    .keySpec(materials.algorithmSuite().dataKeyAlgorithm() + "_" + materials.algorithmSuite().dataKeyLengthBits())
```

Comment:
nit: this works, we can do better by validating against the SDK's `DataKeySpec` enum: 

```
DataKeySpec dataKeySpec;
if (!materials.algorithmSuite().dataKeyAlgorithm().equals("AES")) {
  throw new S3EncryptionClientException(String.format("The data key algorithm %s is not supported by AWS " +
          "KMS", materials.algorithmSuite().dataKeyAlgorithm()));
}
switch (materials.algorithmSuite().dataKeyLengthBits()) {
  case 128:
      dataKeySpec = DataKeySpec.AES_128;
      break;
  case 256:
      dataKeySpec = DataKeySpec.AES_256;
      break;
  default:
      throw new S3EncryptionClientException(String.format("The data key length %d is not supported by " +
              "AWS KMS", materials.algorithmSuite().dataKeyLengthBits()));
}
```

Reply:
I guess we do not use, `AES_128` to generate data key in our algorithm suites. But, I agree with the adding validation.